File: Leetcode_with_Cpp/447numberOfBoomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int res = 0;
        for(int i = 0; i < points.size(); i++) {        //选择第一个点
            unordered_map<int, int> um;                 //距离, 个数
            for(int j = 0; j < points.size(); j++) {    //选择第二个点
                if(i == j) continue;                    //如果i和j指向同一点, 跳过
                int dist2 = (points[i][0] - points[j][0]) * (points[i][0] - points[j][0]) + 
                            (points[i][1] - points[j][1]) * (points[i][1] - points[j][1]);  //计算距离
                um[dist2] ++;
            }
            for(auto p : um) {
                res += p.second * (p.second - 1);
            }
        }
        return res;
    }
};

File: py/447numberOfBoomerangs.py
class Solution(object):
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        res = 0
        for i in points:
            dic = {}
            for j in points:
                distance = (i[0] - j[0])**2 + (i[1] - j[1])**2
                if distance not in dic:
                    dic[distance] = 1
                else:
                    dic[distance] += 1
            for val in dic.values():
                if val >= 2:
                    res += val*(val-1)
        return res